pkgs/http/lib/src/client.dart

```diff
@@ -210,6 +210,13 @@ Client? get zoneClient {
 ///  runWithClient(() => runApp(const MyApp()), clientFactory);
 /// }
 /// ```
+///
+/// If [runWithClient] is used and the environment defines
+/// `no_default_http_client=true` then generated binaries may be smaller e.g.
```

Comment:
Can we expand this? I would also describe the risk of using `no_defualt_http_client=true` any usage of `Client()` outside of this zone will throw.

Reply:
Done